### ISSUE TITLE
Consider box rather than lines in workbook response areas

### ISSUE THREAD

**Issue by er1c456** [bootstrapworld/curriculum#1847]
#1798 @retabak @flannery-denny @schanzer
I'm beginning to review workbooks in Algebra materials and experienced a visceral reaction to the first pages of prompts. I was brought back into my classroom, bringing myself to review student writings, the motion detector light goes out on me again. 

My students struggled to use lined response format. Many lacked penmanship to use lines effectively and others needed to express themselves more creatively. I also needed to use handouts in both a printed and online setting. Lines in docs just don't work for response entry. Blank lines also add a good deal of noise to the page for the easily distracted.

I know lines in the workbook have been around for a while. I might have suggested them. Most likely, you have thought about this and made an informed choice. I don't know whether there is an abstraction to swap between response area formats. 

Please take my experience of stress encountering pages of lines in our workbook with a grain of salt, but if you haven't, please consider the alternative of an open response area or a lighter weight format such as dot grid. Our students who write by hand less and less will thank you.

**Comment by schanzer** [comment]
@er1c456 this is really helpful feedback, and an interesting proposal! Could you screenshot an example of a page where you feel they do a particularly terrific job?

**Comment by er1c456** [comment]
@schanzer Why don't I see a "convert to google docs" when I open [this lesson](https://bootstrapworld.org/materials/fall2024-beta/en-us/lessons/computing-needs-all-voices/pages/windows-mirrors.html)? Why don't I see more materials at that link or in my own build?

Why don't I see a convert to docs button when I view pages from [Spring 2024](https://bootstrapworld.org/materials/spring2024/en-us/lessons/computing-needs-all-voices/pages/NoticeAndWonder.html)?

I'd need to work in a gdocs to demonstrate.

**Comment by schanzer** [comment]
@er1c456 We no long support save to drive

**Comment by flannery-denny** [comment]
@er1c456 Clarifying your feedback. Would your ideal just be blank space or something else?

I've never liked the lines, find them overstimulating, and pushed back on them pretty hard a while back. In the past, ES argued adamantly that I am an outlier (he's not entirely wrong about this on some counts ;) for preferring blank space and that teachers feel students need lines. 

@schanzer I'm not sure where the line weights are at these days (a lot of css rules have changed this year), but at some point we had compromised and made the lines a light grey. Let's at least make sure that's still the case.

**Comment by er1c456** [comment]
@schanzer I'm reeling from news of the loss of convert to docs. I'm sure there is a good reason, but it's a blow.


> @er1c456 Clarifying your feedback. Would your ideal just be blank space or something else? ie do you really think boxes would be preferable on the page you linked or would your optimal just be space to write?

I do think some kind of structure is important. I have certainly used box more than any other format: typically shaded or highlighted depending on if it's print or online only. I like dot grid whenever some kind of diagram could be called for. I could also imagine a single vertical line or tick marks on the left or two sides of the box.


> I've never liked the lines, find them overstimulating, and pushed back on them pretty hard a while back. In the past, ES argued adamantly that I am an outlier (he's not entirely wrong about this on some counts ;) for preferring blank space and that teachers feel students need lines.
> 
> @schanzer I'm not sure where the line weights are at these days (a lot of css rules have changed this year), but at some point we had compromised and made the lines a light grey. Let's at least make sure that's still the case.

The Regents definitely makes them write on lines. Most teachers I knew didn't want to stray far from the test format.  I still agree that something lighter than lines or very faint lines would be preferable.

I was expecting you all had thought about this. Thanks for considering.

**Comment by schanzer** [comment]
@er1c456 You deserve an explanation. 

Ending GDocs support was not an easy decision. It was something we'd be wrestling with for the last 2 years, ever since we returned to in-person instruction. But a number of factors fed into it:

1. GDocs HTML import feature was incredibly limited, and made importing our stuff literally impossible. In fact, we had to maintain _two separate copies of every single lesson, workbook page, etc_ - one that used modern HTML and another than stuck strickly to IE5-level features (yes, IE 5.0!). A lot of the reason our workbook pages look SO much better than they did a few years ago is that we've begun using modern HTML/CSS features like `display:flex`, pseudo-selectors, etc. None of these work in GDocs, so as our pages got better their GDoc equivalents began to look worse and worse by comparison. The code to support GDocs was growing in complexity, and every new feature we wanted to use had to be weighed against the cost of a "good enough" gdoc equivalent.
2. Even with that, the import feature did not support images. As our lessons and wbpages became richer and include things like charts and graphs, the delta between a printed page and a gdoc page grew larger and larger.
3. The impact of 1+2 is that teachers were complaining more and more about "Save to Drive not working", because they (reasonably) had an expectation that their GDocs would look at least a little like their web equivalents. And because of 1+2, we knew that we could _never_ live up to their expectations, and this would forever be a point of frustration for users.
4. GDoc support was created at the start of the pandemic, when everyone was teaching remotely and there were no good tools for editing PDFs on the computer. Since then, the need for an all-digital solution has dropped enormously. And at the same time, there's been an explosion of tools that let students edit PDFs without having to print them. 

In summary - our solution was _never_ perfect, but over time it was increasingly imperfect and preventing us from making our materials better. And at the same time, that solution had radically fewer users and a much better solution existed. Taken together, we just couldn't justify the enormous technical debt required to continue it.

I hope this helps you feel at least a little better about the decision. But if not, know that I'll still pour one out for "Save to GDrive"...
